### PR TITLE
added entity, repository, and migration for recommendation requests, issue #45, #46

### PR DESCRIPTION
**Pull request by skanda-vasishta**
This PR adds an entity, repository, and database migration file for recommendation requests, which will allow for database support for making recommendation requests. Through the H2 console, the user will be able to view recommendation request database

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java
Comment (SexyCarrots) [inline src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java:1]: LGTM

File: frontend/package-lock.json
Comment (SexyCarrots) [inline frontend/package-lock.json:1]: LGTM dependency

File: src/main/java/edu/ucsb/cs156/rec/repositories/RecommendationRequestRepository.java
Comment (SexyCarrots) [inline src/main/java/edu/ucsb/cs156/rec/repositories/RecommendationRequestRepository.java:1]: LGTM

**Review by SexyCarrots — APPROVED** [review]

**Review by SexyCarrots — DISMISSED** [review]
# LGTM

File: src/main/resources/db/migration/changes/RecommendationRequest.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                    "column": {
+                      "name": "RECOMMENDATION_TYPES",
+                      "type": "VARCHAR(255)"
+                    }
+                  },
+                  {
+                    "column": {
+                      "name": "OTHER",
+                      "type": "VARCHAR(255)"
+                    }
```
Comment (pconrad) [inline src/main/resources/db/migration/changes/RecommendationRequest.json:76]: See comments above about this field.

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+public class RecommendationRequest {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  private String professorName;
+  private String professorEmail;
+  private String requesterEmail;
+  private String recommendationTypes; //dropdown?
+  private String other;
```
[the remark this answers is not in the document]
Comment (skanda-vasishta) [reply src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java:33]: Ah, I think it's  redundant to the "details" field, which is where the user would elaborate if they selected other as the recommendation type. I'll delete that.

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+@Entity(name = "recommendationrequest")
+public class RecommendationRequest {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  private String professorName;
+  private String professorEmail;
+  private String requesterEmail;
+  private String recommendationTypes; //dropdown?
```
[the remark this answers is not in the document]
Comment (skanda-vasishta) [reply src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java:32]: sounds good, I'll leave a comment there and keep that in mind.

File: src/main/resources/db/migration/changes/UCSBDates.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
                       "autoIncrement": true,
                       "constraints": {
                         "primaryKey": true,
-                        "primaryKeyName": "CONSTRAINT_5"
+                        "primaryKeyName": "CONSTRAINT_8"
```
[the remark this answers is not in the document]
Comment (skanda-vasishta) [reply src/main/resources/db/migration/changes/UCSBDates.json]: 👍 changed to UCSBDates_PK

File: src/main/resources/db/migration/changes/RecommendationRequest.json (hunk 1 above)
Comment (skanda-vasishta) [reply src/main/resources/db/migration/changes/RecommendationRequest.json]: deleted other field.

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (skanda-vasishta) [reply src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java]: deleted 'other' field.

**Review by michaelljiang — DISMISSED** [review]
LGTM

**Review by michaelljiang — DISMISSED** [review]
LGTM good change for the dates and adding status

**Review by michaelljiang — APPROVED** [review]
Good job catching requesterEmail

**Pull request closed without merging**